### PR TITLE
docs: fill out derp documentation

### PR DESCRIPTION
**Pull request by ramfox**

### CODE REVIEW

File: iroh-net/src/hp/derp.rs [hunk 1]
```diff
@@ -9,9 +9,12 @@
 //! between very aggressive NATs, firewalls, no IPv6, etc? Well, DERP.
 //! Based on tailscale/derp/derp.go
 
+#![deny(missing_docs, rustdoc::broken_intra_doc_links)]
 pub(crate) mod client;
 pub(crate) mod client_conn;
 pub(crate) mod clients;
+/// An http specific Derp Client and Derp Server. Allows for using tls or non tls connection
+/// upgrades.
```
Comment (dignifiedquire) [inline iroh-net/src/hp/derp.rs:17]: I think it is nicer to put the module documentation at the beginning of the file using `//!`

File: iroh-net/src/hp/derp/http.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 mod client;
-mod mesh_clients;
+pub(crate) mod mesh_clients;
```
Comment (dignifiedquire) [inline iroh-net/src/hp/derp/http.rs:2]: why this change?

File: iroh-net/src/hp/derp/http/client.rs [hunk 3]
```diff
@@ -37,46 +37,67 @@ const CONNECT_TIMEOUT: Duration = Duration::from_secs(10);
 const MESH_CLIENT_REDIAL_DELAY: Duration = Duration::from_secs(5);
 
 #[derive(Debug, thiserror::Error)]
+/// Possible connection errors on the [Client]
 pub enum ClientError {
```
Comment (dignifiedquire) [inline iroh-net/src/hp/derp/http/client.rs:41]: I am sorry, but could we unify these to be in the order 1. docs, 2. derive/macros, 3. definition?

```rust
/// Possible connection errors on the [Client]
#[derive(Debug, thiserror::Error)]
pub enum ClientError {
```

File: iroh-net/src/hp/derp/types.rs [hunk 4]
```diff
@@ -101,7 +101,13 @@ impl ServerInfo {
     }
 }
 
+/// A PacketForwarder is can forward a packet to the `dstkey` from the `srckey`.
```
Comment (dignifiedquire) [inline iroh-net/src/hp/derp/types.rs:104]: "is can"?

File: iroh-net/src/hp/derp/http.rs (hunk 2 above)
Comment (ramfox) [reply iroh-net/src/hp/derp/http.rs]: ahh, that was a mistake, changed it back, ty

**Review by dignifiedquire — APPROVED** [review]
